patches/libuv/0001-Add-Managarm-support.patch

```diff
@@ -136,110 +194,192 @@ index 14d5f0c0..598b22c7 100644
-+#if defined(__NR_preadv) && !defined(__managarm__)
-   return syscall(__NR_preadv, fd, iov, iovcnt, (long)offset, (long)(offset >> 32));
-+#elif defined(__managarm__)
++}
++
++
++ssize_t uv__preadv(int fd, const struct iovec *iov, int iovcnt, int64_t offset) {
 +  return preadv(fd, iov, iovcnt, (long)offset, (long)(offset >> 32));
- #else
-   return errno = ENOSYS, -1;
- #endif
-@@ -181,8 +189,10 @@ ssize_t uv__preadv(int fd, const struct iovec *iov, int iovcnt, int64_t offset)
- 
- 
- ssize_t uv__pwritev(int fd, const struct iovec *iov, int iovcnt, int64_t offset) {
--#if defined(__NR_pwritev)
-+#if defined(__NR_pwritev) && !defined(__managarm__)
-   return syscall(__NR_pwritev, fd, iov, iovcnt, (long)offset, (long)(offset >> 32));
-+#elif defined(__managarm__)
++}
++
++
++ssize_t uv__pwritev(int fd, const struct iovec *iov, int iovcnt, int64_t offset) {
 +  return pwritev(fd, iov, iovcnt, (long)offset, (long)(offset >> 32));
- #else
-   return errno = ENOSYS, -1;
- #endif
-@@ -190,8 +200,10 @@ ssize_t uv__pwritev(int fd, const struct iovec *iov, int iovcnt, int64_t offset)
- 
- 
- int uv__dup3(int oldfd, int newfd, int flags) {
--#if defined(__NR_dup3)
-+#if defined(__NR_dup3) && !defined(__managarm__)
-   return syscall(__NR_dup3, oldfd, newfd, flags);
-+#elif defined(__managarm__)
++}
++
++
++int uv__dup3(int oldfd, int newfd, int flags) {
 +  return dup3(oldfd, newfd, flags);
- #else
-   return errno = ENOSYS, -1;
- #endif
-@@ -206,7 +218,7 @@ uv__fs_copy_file_range(int fd_in,
-                        size_t len,
-                        unsigned int flags)
- {
--#ifdef __NR_copy_file_range
-+#if defined __NR_copy_file_range && !defined(__managarm__)
-   return syscall(__NR_copy_file_range,
-                  fd_in,
-                  off_in,
-@@ -228,7 +240,7 @@ int uv__statx(int dirfd,
-   /* __NR_statx make Android box killed by SIGSYS.
-    * That looks like a seccomp2 sandbox filter rejecting the system call.
-    */
--#if defined(__NR_statx) && !defined(__ANDROID__)
-+#if defined(__NR_statx) && !defined(__ANDROID__) && !defined(__managarm__)
-   return syscall(__NR_statx, dirfd, path, flags, mask, statxbuf);
- #else
-   return errno = ENOSYS, -1;
-@@ -237,8 +249,10 @@ int uv__statx(int dirfd,
- 
- 
- ssize_t uv__getrandom(void* buf, size_t buflen, unsigned flags) {
--#if defined(__NR_getrandom)
-+#if defined(__NR_getrandom) && !defined(__managarm__)
-   return syscall(__NR_getrandom, buf, buflen, flags);
-+#elif defined(__managarm__)
++}
++
++
++ssize_t
++uv__fs_copy_file_range(int fd_in,
++                       ssize_t* off_in,
++                       int fd_out,
++                       ssize_t* off_out,
++                       size_t len,
++                       unsigned int flags)
++{
++  return errno = ENOSYS, -1;
++}
++
++
++int uv__statx(int dirfd,
++              const char* path,
++              int flags,
++              unsigned int mask,
++              struct uv__statx* statxbuf) {
++  return errno = ENOSYS, -1;
++}
```

Comment:
This function is only ever called if `defined(__linux__)` (`src/unix/fs.c`, line 1379) and can be removed